### DISCUSSION TITLE
Setting Decimals in Visualization

### DISCUSSION THREAD

**Discussion by wavoigt** [GyroGearl00se/ESPHome-Froeling-Lambdatronic_3200#36] [General]
How can i set the number of decimals in the visualization (in dashboard .yaml or .svg file)?

**Comment by wavoigt** [comment]
no idea?

**Comment by GyroGearl00se** [comment]
Hi @wavoigt,
can you make an example of an value ?

What is the current value and what is your expected output ?

So far I can imagine using filters in esphome like "accuracy_decimals" or "multiply".
For example converting 1000 kg to 1t could be done with multiply by 0.001 if I'm not mistaken.

**Comment by wavoigt** [comment]
It is a issue in my aio visualisation: https://github.com/GyroGearl00se/ESPHome-Froeling-Lambdatronic_3200/discussions/15#discussioncomment-8244849
I get the entities of the Landis+Gyr smartmeter by a HMIP Interface.
The visualisation shows me 123456.0 where I want it to display 123456 without decimals.
I tried it with  <pre>service_data: ${entity.state | int}</pre> but this did not work.
By digging in net i found some possible solutions, but i could not yet test it:
<pre>service_data: ${entity.state.substring(0, entity.state.indexOf("."))}</pre>
or:
<pre>service_data: 
  class "${entity.state.substring(0, entity.state.indexOf('.'))}"</pre>
What do you think?

**Comment by GyroGearl00se** [comment]
Hi @wavoigt,
I just noticed that this remained uncommented. Sorry for that.

Not sure if you've already found a solution. If not, here some possible solution for your request:

You can use Math.round
service_data: '${Math.round(entity.state)}'
This lead to following result:
12.4 -> 12
12.5 -> 13

Or if you just want to "cut off" any decimal, you could also use the Math.floor function.
service_data: '${Math.floor(entity.state)}'

12.1 -> 12
12.9 -> 12

**Comment by wavoigt** [comment]
Tested solution:
service_data: ${entity.state.substring(0, entity.state.indexOf("."))}
is working fine.
